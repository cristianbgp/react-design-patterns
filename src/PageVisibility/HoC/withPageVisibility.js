import React from "react";

function withPageVisibility(Component) {
  function PageVisibility() {
    const [status, setStatus] = React.useState(true);

    React.useEffect(() => {
      function handleVisibility(event) {
        if (document.hidden) {
          console.log("Page is not visible");
          setStatus(false);
        } else {
          console.log("Page is visible");
          setStatus(true);
        }
      }
      document.addEventListener("visibilitychange", handleVisibility);
      return () => {
        window.removeEventListener("visibilitychange", handleVisibility);
      };
    }, [setStatus]);

    return <Component status={status} />;
  }
  return PageVisibility;
}

export default withPageVisibility;
