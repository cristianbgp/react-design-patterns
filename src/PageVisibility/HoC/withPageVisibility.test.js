import React from "react";
import { render } from "@testing-library/react";
import withPageVisibility from "./withPageVisibility";

test("Page Visibility with HoC", () => {
  function Component({ status }) {
    return (
      <>
        <h1>Using High order Components</h1>
        <p>
          Page Visibility: <span>{status.toString()}</span>
        </p>
      </>
    );
  }

  const App = withPageVisibility(Component);
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
