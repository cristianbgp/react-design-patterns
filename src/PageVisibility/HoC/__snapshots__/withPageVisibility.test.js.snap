// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Visibility with HoC 1`] = `
<DocumentFragment>
  <h1>
    Using High order Components
  </h1>
  <p>
    Page Visibility: 
    <span>
      true
    </span>
  </p>
</DocumentFragment>
`;
